$MirOS: ports/www/firesomething/patches/patch-directory_c-sdk_configure_in,v 1.2 2006/05/11 20:55:43 bsiegert Exp $
--- directory/c-sdk/configure.in.orig	2007-07-11 22:43:19.000000000 +0200
+++ directory/c-sdk/configure.in	2008-05-20 20:59:19.000000000 +0200
@@ -42,7 +42,7 @@ dnl ***** END LICENSE BLOCK *****
 AC_PREREQ(2.12)
 AC_INIT(ldap/build/dirver.c)
 
-AC_CONFIG_AUX_DIR(${srcdir}/config/autoconf)
+AC_CONFIG_AUX_DIR(config/autoconf)
 AC_CANONICAL_SYSTEM
 
 dnl ========================================================
@@ -1573,7 +1573,23 @@ mips-sony-newsos*)
     _DEBUG_FLAGS="-gstabs"
 	;;
 
-*-openbsd*)
+*-midnightbsd*)
+    if test -z "$USE_NSPR_THREADS"; then
+        USE_PTHREADS=1
+    fi
+    AC_DEFINE(XP_UNIX)
+    AC_DEFINE(FREEBSD)
+    AC_DEFINE(HAVE_BSD_FLOCK)
+    CFLAGS="$CFLAGS $(DSO_CFLAGS) -ansi -Wall"
+    MOZ_OBJFORMAT=elf
+    DLL_SUFFIX=so
+    DSO_CFLAGS=-fPIC
+    DSO_LDOPTS=-Bshareable
+    MDCPUCFG_H=_freebsd.cfg
+    PR_MD_CSRCS=freebsd.c
+    ;;
+
+*-openbsd*|*-mirbsd*)
     LD='$(CC)'
     AC_DEFINE(XP_UNIX)
     AC_DEFINE(OPENBSD)
@@ -2180,7 +2196,7 @@ if test -n "$USE_PTHREADS"; then
             _PTHREAD_LDFLAGS=
         fi
 	    ;;
-    *-freebsd*|*-openbsd*|*-bsdi*|*-netbsd*)
+    *-freebsd*|*-openbsd*|*-bsdi*|*-netbsd*|*-mirbsd*|*-midnightbsd*)
 	    AC_DEFINE(_THREAD_SAFE)
 	    dnl -pthread links in -lc_r, so don't specify it explicitly.
 	    if test "$ac_cv_have_dash_pthread" = "yes"; then
